spec/models/source_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Source, type: :model do
+  context 'when saving two Sources with different cases' do
+    it 'saves only one Source' do
+      described_class.find_or_create_by({ name: 'nimble' })
+
+      expect do
+        described_class.find_or_create_by({ name: 'nImBle' })
+      end.not_to change(described_class, :count)
+    end
+  end
+
+  context 'when saving two Sources with different names' do
```

Comment:
```suggestion
  context 'when creating two Sources with different names' do
```

We can `save` a change (e.g. when updating the values of attributes) without changing the count 😇